src/contract.ts

```diff
@@ -0,0 +1,22 @@
+import { ContractAction } from "./types.js";
+
+export function extractActionsFromContract(code: string): ContractAction[] {
+  const regex = /actions\s+triggered_by:\s+transaction,\s+on:\s+([\w\s.,()]+?)\s+do/g;
+
+  let actions = [];
+  for (const match of code.matchAll(regex)) {
+    const fullAction = match[1];
+
+    const regexActionName = /(\w+)\((.*?)\)/g;
+    for (const actionMatch of fullAction.matchAll(regexActionName)) {
+      const name = actionMatch[1];
+      const parameters = actionMatch[2] != "" ? actionMatch[2].split(",") : [];
```

Comment:
Here we can have some weird behavior. You can try the contract 00002075E9FB5329C7F5149802EB75F925D79B5FEC5BFCF09DC81993CB9C39FEAF5D on the mainnet for example.
Some parameters are not used in the action, and so they are replaced by `_`. So in the UI the parameters name doesn't mean anything.
Maybe you could also retrieve the parameters from the condition, and for each parameter index, take the one that does not start by a `_`

Reply:
Yes we noticed that, I can try with condition, but in general case, it would be better to not omit variable name even if not used.

Reply:
This would need a rework on the blockchain, and need to be retrocompatible. This is not something we can do quickly I guess

Reply:
My point was more a best practice for external tooling than rework on smart contract, best practice to put name variables otherwise tooling might not work as expected.